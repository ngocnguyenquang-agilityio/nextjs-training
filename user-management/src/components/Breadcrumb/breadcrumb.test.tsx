import { fireEvent, render } from '@testing-library/react';

// Components
import { Breadcrumb } from '.';
import React, { type ReactElement } from 'react';

const props = {
  breadcrumbs: [
    {
      label: 'Home',
      href: '/home'
    },
    {
      label: 'Users',
      href: '/home/users'
    }
  ]
};

describe('Breadcrumb component', () => {
  jest.mock(
    'next/link',
    () =>
      ({ children, ...rest }: { children: ReactElement }) =>
        React.cloneElement(children, { ...rest })
  );

  test('Should render successfully', () => {
    const comp = render(<Breadcrumb {...props} />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<Breadcrumb {...props} />);
    expect(comp).toMatchSnapshot();
  });

  test('Should render correct active breadcrumb', () => {
    const comp = render(<Breadcrumb {...props} />);
    const breadcrumb = comp.queryByTestId('link-Users');
    expect(breadcrumb?.getAttribute('class')).toMatch(/text-blue-500/gi);
  });

  test('Should got correct href when click', async () => {
    const comp = render(<Breadcrumb {...props} />);
    const link = comp.queryByTestId('link-Home')?.querySelector('a') as HTMLAnchorElement;
    await fireEvent.click(link);
    expect(link.getAttribute('href')).toBe('/home');
  });
});
