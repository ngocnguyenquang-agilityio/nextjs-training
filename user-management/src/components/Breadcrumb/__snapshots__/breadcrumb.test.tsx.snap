// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="Breadcrumb"
        class="mb-6 block"
      >
        <ol
          class="flex text-xl md:text-xl"
        >
          <li
            class="text-gray-400"
            data-testid="link-Home"
          >
            <a
              class="hover:text-blue-400"
              href="/home"
            >
              Home
            </a>
            <span
              class="mx-3 inline-block"
            >
              /
            </span>
          </li>
          <li
            class="text-blue-500"
            data-testid="link-Users"
          >
            <a
              class="hover:text-blue-400"
              href="/home/users"
            >
              Users
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="Breadcrumb"
      class="mb-6 block"
    >
      <ol
        class="flex text-xl md:text-xl"
      >
        <li
          class="text-gray-400"
          data-testid="link-Home"
        >
          <a
            class="hover:text-blue-400"
            href="/home"
          >
            Home
          </a>
          <span
            class="mx-3 inline-block"
          >
            /
          </span>
        </li>
        <li
          class="text-blue-500"
          data-testid="link-Users"
        >
          <a
            class="hover:text-blue-400"
            href="/home/users"
          >
            Users
          </a>
        </li>
      </ol>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
